.pre-commit-config.yaml

```diff
@@ -0,0 +1,27 @@
+exclude: '.git'
+default_stages: [commit]
+
+repos:
+  - repo: https://github.com/psf/black
+    rev: 21.12b0
+    hooks:
+      - id: black
+
+  - repo: https://github.com/timothycrosley/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.9.2
+    hooks:
+      - id: flake8
+        additional_dependencies: [flake8-isort, flake8-bugbear]
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```

Comment:
Selected [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks) actions:
- trailing-whitespace: trims all whitespace at the end of lines
- end-of-file-fixer: ensures files end with a newline
- debug-statements: checks for leftover debug statements in the code
- double-quote-string-fixer: changes all single quotes to double quotes